internal/diff_str.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+import type { DiffResult } from "./_types.ts";
+import { _internals } from "./_internals.ts";
```

Comment:
Looks like this can be replaced by importing `diff` from `diff.ts`? (and we can remove `_internals.ts`)